src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const { id, title, completed, userId } = selectedTodo;
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(userId)
+      .then(setUser)
+      .finally(() => setIsLoading(false));
+  }, [userId]);
+
+  const handleCloseModal = () => {
+    setSelectedTodo(null);
+  };
 
-export const TodoModal: React.FC = () => {
   return (
```

Comment:
You are interacting with the DOM directly by using className to define classes. It is recommended to use the [classnames library](https://www.npmjs.com/package/classnames) for defining classes conditionally in React.